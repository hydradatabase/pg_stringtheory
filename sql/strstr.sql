CREATE EXTENSION stringtheory;

-- no match
SELECT stringtheory.strstr('hello', 'world');

-- match with 0
SELECT stringtheory.strstr('hello', 'hello');

-- match on a 16 byte boundary
SELECT stringtheory.strstr('1234567890123456', '1234567890123456');

-- match when partial
SELECT stringtheory.strstr('123456', '12345');

-- needle found in haystack
SELECT stringtheory.strstr('hello world', 'ello');

-- haystack in needle not found
SELECT stringtheory.strstr('ello', 'hello world');
