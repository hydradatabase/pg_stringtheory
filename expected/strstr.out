CREATE EXTENSION IF NOT EXISTS stringtheory;
NOTICE:  extension "stringtheory" already exists, skipping
-- no match
SELECT stringtheory.strstr('hello', 'world');
 strstr 
--------
     -1
(1 row)

-- match with 0
SELECT stringtheory.strstr('hello', 'hello');
 strstr 
--------
      0
(1 row)

-- match on a 16 byte boundary
SELECT stringtheory.strstr('1234567890123456', '1234567890123456');
 strstr 
--------
      0
(1 row)

-- match when partial
SELECT stringtheory.strstr('123456', '12345');
 strstr 
--------
      0
(1 row)

-- needle found in haystack
SELECT stringtheory.strstr('hello world', 'ello');
 strstr 
--------
      1
(1 row)

-- haystack in needle not found
SELECT stringtheory.strstr('ello', 'hello world');
 strstr 
--------
     -1
(1 row)

-- test strstr in a CTE
WITH a AS (SELECT md5(generate_series(1, 1000)::text) b)
SELECT COUNT(*) FROM a
WHERE stringtheory.strstr(b, '00') >= 0;
 count 
-------
   114
(1 row)

-- test strstr in a table
CREATE TEMPORARY TABLE stringtheory_test
(a text);
INSERT INTO stringtheory_test
SELECT md5(generate_series(1,1000)::text);
SELECT COUNT(*) FROM stringtheory_test
WHERE stringtheory.strstr(a, '00') >= 0;
 count 
-------
   114
(1 row)

DROP TABLE stringtheory_test;
